docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+`INPUT_KEY \in Int`. The reason is that this expression requires the model
+checker to reason about one integer, though it ranges over the infinite set of
+integers.
+
+**Value generators.** Fortunately, this problem can be easily circumvented by
+using Apalache [Value generators][], which are inspired by property-based testing,
+e.g., [QuickCheck][].
+
+Let us rewrite `ConstInit` in `MC5_8.tla` as follows:
+
+```tla
+ConstInit ==
+    /\ INPUT_KEY = Gen(1)
+    /\ INPUT_SEQ = Gen(MAX_INT)
+```
+
+In this new version, we use value generators to:
+
+  - produce an unrestricted integer to be used as a value of `INPUT_KEY` and
+  - produce a sequence of integers to be used as a value of `INPUT_SEQ`. This
+    sequence is unrestricted, except its length is bounded with `MAX_INT`,
+    which is exactly what we need in our case study.
+
+In contrast to property-based testing, an Apalache generator is not randomly
+producing values. Rather, it simply introduces a data structure whose size is
+bounded with the argument of `Gen`. This lets Apalache check all instances of
+the data structure, without enumerating the instances!
+
+By doing so, we are able to check the specification for all the inputs, when we
+fix the bit width. In the model `MC5_8.tla`, we fix `INT_WIDTH` to 8, to get
+the feedback fast.
+
+Let us check `Postcondition` again:
+
+```sh
+$ apalache-mc check --cinit=ConstInit --inv=Postcondition MC5_8.tla
+...
+State 2: state invariant 0 violated. Check the counterexample in: 
+  /[a long path]/counterexample1.tla
+...
+```
+
+Let us inspect the counterexample:
+
+```tla
+---------------------------- MODULE counterexample ----------------------------  
+EXTENDS MC5_8
+
+(* Constant initialization state *)
+ConstInit == INPUT_KEY = -1 /\ INPUT_SEQ = <<0, -1>>
+
+(* Initial state *)
+State0 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = 1
+    /\ isTerminated = FALSE
+    /\ low = 0
+    /\ returnValue = 0
+
+(* Transition 1 to State1 *)
+State1 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = -1
+    /\ isTerminated = FALSE
+    /\ low = 0
+    /\ returnValue = 0
+
+(* Transition 3 to State2 *)
+State2 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = -1
+    /\ isTerminated = TRUE
+    /\ low = 0
+    /\ returnValue = -1
+...
+```
+
+**Fixing Postcondition.** Is it a real issue? It is, but it is not the issue of
+the search, rather our invariant `Postcondition` is imprecise.
+
+## Step 5b: Fixing the postcondition
+
+If we check our source of truth, that is, the Java documentation in
+[Arrays.java in OpenJDK][], we will see the following sentences:
+
+    The range must be sorted (as by the {@link #sort(int[], int, int)} method)
+    prior to making this call. If it is not sorted, the results are undefined.
+    If the range contains multiple elements with the specified value, there is
+    no guarantee which one will be found.
```

Comment:
```suggestion
```

You're only specifying sortedness below; may want to omit this sentence.

Reply:
Well, this is the sentence from the Java doc. I would keep it as is.

Reply:
You're right, sorry. I misread the English spec...